frontend/src/store/wallet.ts

```diff
@@ -1,26 +1,40 @@
 import { computed, onMounted, ref, watch } from '@vue/composition-api';
-import Onboard from 'bnc-onboard';
-import { API as OnboardAPI } from 'bnc-onboard/dist/src/interfaces';
-import { KeyPair, Umbra, StealthKeyRegistry, utils } from '@umbra/umbra-js';
 
+import Onboard, { OnboardAPI } from '@web3-onboard/core';
+import injectedModule from '@web3-onboard/injected-wallets';
+import walletConnectModule from '@web3-onboard/walletconnect';
+import coinbaseWalletModule from '@web3-onboard/coinbase';
+import ledgerModule from '@web3-onboard/ledger';
+import trezorModule from '@web3-onboard/trezor';
+
+import { KeyPair, Umbra, StealthKeyRegistry, utils } from '@umbra/umbra-js';
 import {
   Chain,
   MulticallResponse,
   NATIVE_TOKEN_ADDRESS,
   Network,
   Provider,
   Signer,
+  supportedChains,
   supportedChainIds,
   SupportedChainId,
   TokenInfoExtended,
 } from 'components/models';
 import { formatNameOrAddress, lookupEnsName, lookupCnsName } from 'src/utils/address';
-import { ERC20_ABI, MAINNET_PROVIDER, MAINNET_RPC_URL, MULTICALL_ABI, MULTICALL_ADDRESSES } from 'src/utils/constants';
-import { BigNumber, Contract, getAddress, Web3Provider, parseUnits } from 'src/utils/ethers';
+import { ERC20_ABI, MAINNET_PROVIDER, MULTICALL_ABI, MULTICALL_ADDRESSES } from 'src/utils/constants';
+import { BigNumber, Contract, Web3Provider, parseUnits } from 'src/utils/ethers';
 import { UmbraApi } from 'src/utils/umbra-api';
 import { getChainById } from 'src/utils/utils';
 import useSettingsStore from 'src/store/settings';
-
+import enLocal from 'src/i18n/locales/en-us.json';
+import chLocal from 'src/i18n/locales/zh-cn.json';
+
+// Wallet configurations.
+const injected = injectedModule();
+const walletConnect = walletConnectModule();
```

Comment:
Notes on walletconnect. It looks like onboard wallet connect does not support wallet_switchEthereumChain. Looking into potential alternatives.

https://github.com/blocknative/web3-onboard/blob/v2-web3-onboard-develop/packages/walletconnect/src/index.ts#L183

Reply:
Made an [issue](https://github.com/blocknative/web3-onboard/issues/1313) and hope to get this implemented on there side.

Reply:
Awesome thank you